language_formatters_pre_commit_hooks/pretty_format_java.py

```diff
@@ -130,6 +187,51 @@ def pretty_format_java(argv: typing.Optional[typing.List[str]] = None) -> int:
                 ", ".join(output.splitlines()),
             ),
         )
+    return 0 if status == 0 else 1
+
+
+def format_palantir(args):
+    # Palantir Java Formatter starting from 2.0.0 does not support Java 1.8,
+    # before that version the tool can only be executed on Java up to version 20.
+    # Context: https://github.com/palantir/palantir-java-format/releases/tag/v2.0.0
+    if Version(args.palantir_java_formatter_version) >= Version("2.0.0"):
+        assert_min_jdk_version(Version("11.0"))  # pragma: no cover
+
+    # Palantir Java formatter is not compatible with Java 21 and above yet.
+    # https://github.com/palantir/palantir-java-format/issues/977 contains
+    # more links for Java compatibility
+    assert_max_jdk_version(Version("21.0"))  # pragma: no cover
```

Comment:
nit: If you have any test that would try this logic this line will be covered. No reason to exclude it for coverage

```suggestion
    assert_max_jdk_version(Version("21.0"))
```

Reply:
I've covered this line, but lines below I can't cover on Java 21. Do you have any suggestions?